approval/repository.go

```diff
@@ -14,9 +21,22 @@ func NewRepository(db *gorm.DB) *repository {
 	return &repository{db}
 }
 
-func (r *repository) GetPendingApprovals(approverEmail string) ([]*domain.Approval, error) {
+func (r *repository) ListApprovals(filters map[string]interface{}) ([]*domain.Approval, error) {
```

Comment:
was thinking if it is better if `ListApprovals` is accepting `ListFilter` struct as the parameter.
```
func (r *repository) ListApprovals(filter ListFilter) ([]*domain.Approval, error)

// using it
repo.ListApprovals(repo.ListFilter{
  User: req.User(),
  Statuses: req.GetStatuses(),
})
```

wdyt?